### PR TITLE
Rh/UI improvements

### PR DESCRIPTION
**Pull request by laholstege**
image
image
image

send on enter, new question button (not sure if it should still be a + next to the chat?), default question buttons, side links (that dont go anywhere yet) 
organizing files for SSR and readability
still needs loading animations/transitions, going to add that in the next PR since that will be using suspense & sending from the server 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new layout component for the login page for enhanced user experience.
	- Added a redirection functionality to the root page from the new page.
	- Launched a chat interface with features for message interaction and predefined questions.
	- Updated the homepage to include a disclaimer for informational purposes.
	- Enhanced the chat interface with a start screen feature.
	- Improved sidebar navigation with new links and user name display adjustments.

- **Refactor**
	- Reorganized the login component for better form submission handling.
	- Adjusted the main layout to include a new container class for better content organization.
	- Reordered import statements in the authentication module for consistency.

- **Style**
	- Added styling for the new chat interface components.

- **Chores**
	- Updated footer to use Next.js Image component for the UC Davis logo.

### CODE REVIEW

**Comment by laholstege** [comment]
please be nice to me little rabbit

File: src/components/layout/sidebar.tsx [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
     </div>
   );
 };
 
-const UserNameDisplay = async () => {
-  const session = await auth();
+export default Sidebar;
 
+const UserNameDisplay: React.FC = async ({}) => {
+  const session = (await auth()) as Session;
```
[the remark this answers is not in the document]
Comment (laholstege) [reply src/components/layout/sidebar.tsx:33]: its a server component

[the remark this answers is not in the document]
Comment (laholstege) [reply src/components/layout/sidebar.tsx:33]: but its ok i get confused too

File: src/app/auth/login/page.tsx [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
-    </div>
+          await signIn('azure-ad', {
+            redirectTo: '/',
+          });
+        }}
+      >
+        <button type='submit' className='btn btn-primary btn-lg'>
+          Login
+        </button>
+      </form>
```
[the remark this answers is not in the document]
Comment (laholstege) [reply src/app/auth/login/page.tsx:22]: this is a server action

[the remark this answers is not in the document]
Comment (laholstege) [reply src/app/auth/login/page.tsx:22]: it is a standard pattern in nextjs's App Router
https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations

File: src/app/auth/login/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  return (
+    <div className='card'>
+      <div className='card-body text-center'>{children}</div>
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (laholstege) [reply src/app/auth/login/layout.tsx]: changed this to be LoginLayout. these styles will only apply to the /login page

File: src/components/layout/sidebar.tsx [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
             Policy Wonk <br />
             <span className='subtitle'>Your UC Policy expert</span>
-          </a>
+          </Link>
         </h1>
-
         <div>
           <UserNameDisplay />
         </div>
+        <SidebarLinks />
```
[the remark this answers is not in the document]
Comment (laholstege) [reply src/components/layout/sidebar.tsx:23]: changed SidebarLinks to clarify that <Link> is just for internal links and <a> for external. <Link> does fancy nextjs stuff and will prefetch other routes when the link is in the viewport

**Review by srkirkland — APPROVED** [review]

File: src/components/chat/defaultQuestions.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          disabled={!allowSend}
+        >
+          {question}
+        </button>
+      ))}
+    </div>
+  );
+};
+
+export default DefaultQuestions;
```
[the remark this answers is not in the document]
Comment (laholstege) [reply src/components/chat/defaultQuestions.tsx:37]: i'm leaving it for now

**Pull request merged**